Description: Add support for steelseries Gaming headphones
Origin: upstream, https://cgit.freedesktop.org/pulseaudio/pulseaudio/commit/?id=83675b3745c64bd738400eae44eb4daa195ed88a
Bug-Ubuntu: https://launchpad.net/bugs/1758736
Author: Dave Chiluk <chiluk@ubuntu.com>

-----------------------------------------------
commit 83675b3745c64bd738400eae44eb4daa195ed88a
Author: Bert Hekman <demontpx@gmail.com>
Date:   Tue Feb 13 20:06:12 2018 +0100

    alsa-mixer: add support for SteelSeries Arctis 5 and renamed Arctis 7 files appropriately

Index: pulseaudio-12.2/src/Makefile.am
===================================================================
--- pulseaudio-12.2.orig/src/Makefile.am
+++ pulseaudio-12.2/src/Makefile.am
@@ -1327,7 +1327,8 @@ dist_alsaprofilesets_DATA = \
 		modules/alsa/mixer/profile-sets/native-instruments-korecontroller.conf \
 		modules/alsa/mixer/profile-sets/kinect-audio.conf \
 		modules/alsa/mixer/profile-sets/sb-omni-surround-5.1.conf \
-		modules/alsa/mixer/profile-sets/steelseries-arctis-usb-audio.conf \
+		modules/alsa/mixer/profile-sets/steelseries-arctis-5-usb-audio.conf \
+		modules/alsa/mixer/profile-sets/steelseries-arctis-7-usb-audio.conf \
 		modules/alsa/mixer/profile-sets/dell-dock-tb16-usb-audio.conf
 
 if HAVE_UDEV
@@ -1371,9 +1372,11 @@ dist_alsapaths_DATA = \
 		modules/alsa/mixer/paths/hdmi-output-5.conf \
 		modules/alsa/mixer/paths/hdmi-output-6.conf \
 		modules/alsa/mixer/paths/hdmi-output-7.conf \
-		modules/alsa/mixer/paths/steelseries-arctis-input.conf \
-		modules/alsa/mixer/paths/steelseries-arctis-output-mono.conf \
-		modules/alsa/mixer/paths/steelseries-arctis-output-stereo.conf
+		modules/alsa/mixer/paths/steelseries-arctis-5-output-chat.conf \
+		modules/alsa/mixer/paths/steelseries-arctis-5-output-game.conf \
+		modules/alsa/mixer/paths/steelseries-arctis-7-input.conf \
+		modules/alsa/mixer/paths/steelseries-arctis-7-output-mono.conf \
+		modules/alsa/mixer/paths/steelseries-arctis-7-output-stereo.conf
 
 endif
 
Index: pulseaudio-12.2/src/modules/alsa/alsa-mixer.c
===================================================================
--- pulseaudio-12.2.orig/src/modules/alsa/alsa-mixer.c
+++ pulseaudio-12.2/src/modules/alsa/alsa-mixer.c
@@ -2545,6 +2545,8 @@ static int path_verify(pa_alsa_path *p)
         { "iec958-passthrough-output",  N_("Digital Passthrough (S/PDIF)") },
         { "multichannel-input",         N_("Multichannel Input") },
         { "multichannel-output",        N_("Multichannel Output") },
+        { "steelseries-arctis-5-output-game", N_("Game Output") },
+        { "steelseries-arctis-5-output-chat", N_("Chat Output") },
     };
 
     pa_alsa_element *e;
Index: pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-5-output-chat.conf
===================================================================
--- /dev/null
+++ pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-5-output-chat.conf
@@ -0,0 +1,27 @@
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2.1 of the
+# License, or (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
+
+; Steelseries Arctis 5 USB headset stereo chat path. The headset has two
+; output devices. The first one is meant for voice audio, and the second
+; one meant for everything else. The purpose of this unusual design is to
+; provide separate volume controls for voice and other audio, which can be
+; useful in gaming.
+
+[General]
+priority = 50
+
+[Element Com Speaker]
+switch = mute
+volume = merge
Index: pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-5-output-game.conf
===================================================================
--- /dev/null
+++ pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-5-output-game.conf
@@ -0,0 +1,27 @@
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2.1 of the
+# License, or (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
+
+; Steelseries Arctis 5 USB headset stereo game path. The headset has two
+; output devices. The first one is meant for voice audio, and the second
+; one meant for everything else. The purpose of this unusual design is to
+; provide separate volume controls for voice and other audio, which can be
+; useful in gaming.
+
+[General]
+priority = 99
+
+[Element PCM]
+switch = mute
+volume = merge
Index: pulseaudio-12.2/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules
===================================================================
--- pulseaudio-12.2.orig/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules
+++ pulseaudio-12.2/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules
@@ -105,6 +105,7 @@ ATTRS{idVendor}=="0763", ATTRS{idProduct
 ATTRS{idVendor}=="045e", ATTRS{idProduct}=="02bb", ENV{PULSE_PROFILE_SET}="kinect-audio.conf"
 ATTRS{idVendor}=="041e", ATTRS{idProduct}=="322c", ENV{PULSE_PROFILE_SET}="sb-omni-surround-5.1.conf"
 ATTRS{idVendor}=="0bda", ATTRS{idProduct}=="4014", ENV{PULSE_PROFILE_SET}="dell-dock-tb16-usb-audio.conf"
-ATTRS{idVendor}=="1038", ATTRS{idProduct}=="1260", ENV{PULSE_PROFILE_SET}="steelseries-arctis-usb-audio.conf"
+ATTRS{idVendor}=="1038", ATTRS{idProduct}=="1250", ENV{PULSE_PROFILE_SET}="steelseries-arctis-5-usb-audio.conf"
+ATTRS{idVendor}=="1038", ATTRS{idProduct}=="1260", ENV{PULSE_PROFILE_SET}="steelseries-arctis-7-usb-audio.conf"
 
 LABEL="pulseaudio_end"
Index: pulseaudio-12.2/src/modules/alsa/mixer/profile-sets/steelseries-arctis-5-usb-audio.conf
===================================================================
--- /dev/null
+++ pulseaudio-12.2/src/modules/alsa/mixer/profile-sets/steelseries-arctis-5-usb-audio.conf
@@ -0,0 +1,22 @@
+[General]
+auto-profiles = yes
+
+[Mapping analog-chat]
+description = Chat
+device-strings = hw:%f,0,0
+channel-map = left,right
+paths-input = analog-input-mic
+paths-output = steelseries-arctis-5-output-chat
+
+[Mapping analog-game]
+description = Game
+device-strings = hw:%f,1,0
+channel-map = left,right
+paths-output = steelseries-arctis-5-output-game
+direction = output
+
+[Profile output:analog-chat+output:analog-game+input:analog-chat]
+output-mappings = analog-chat analog-game
+input-mappings = analog-chat
+priority = 5100
+skip-probe = yes
Index: pulseaudio-12.2/src/modules/alsa/mixer/profile-sets/steelseries-arctis-usb-audio.conf
===================================================================
--- pulseaudio-12.2.orig/src/modules/alsa/mixer/profile-sets/steelseries-arctis-usb-audio.conf
+++ /dev/null
@@ -1,43 +0,0 @@
-# This file is part of PulseAudio.
-#
-# PulseAudio is free software; you can redistribute it and/or modify
-# it under the terms of the GNU Lesser General Public License as
-# published by the Free Software Foundation; either version 2.1 of the
-# License, or (at your option) any later version.
-#
-# PulseAudio is distributed in the hope that it will be useful, but
-# WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
-# General Public License for more details.
-#
-# You should have received a copy of the GNU Lesser General Public License
-# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
-
-; Steelseries Arctis 7 USB headset. The headset has a microphone and two output
-; devices. The first output device is mono, meant for voice audio, and the
-; second one is stereo, meant for everything else. The purpose of this unusual
-; design is to provide separate volume controls for voice and other audio,
-; which can be useful in gaming.
-;
-; See default.conf for an explanation on the directives used here.
-
-[General]
-auto-profiles = yes
-
-[Mapping analog-mono]
-device-strings = hw:%f,0,0
-channel-map = mono
-paths-output = steelseries-arctis-output-mono
-paths-input = steelseries-arctis-input
-
-[Mapping analog-stereo]
-device-strings = hw:%f,1,0
-channel-map = left,right
-paths-output = steelseries-arctis-output-stereo
-direction = output
-
-[Profile output:analog-mono+output:analog-stereo+input:analog-mono]
-output-mappings = analog-mono analog-stereo
-input-mappings = analog-mono
-priority = 5100
-skip-probe = yes
Index: pulseaudio-12.2/src/modules/alsa/mixer/profile-sets/steelseries-arctis-7-usb-audio.conf
===================================================================
--- /dev/null
+++ pulseaudio-12.2/src/modules/alsa/mixer/profile-sets/steelseries-arctis-7-usb-audio.conf
@@ -0,0 +1,43 @@
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2.1 of the
+# License, or (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
+
+; Steelseries Arctis 7 USB headset. The headset has a microphone and two output
+; devices. The first output device is mono, meant for voice audio, and the
+; second one is stereo, meant for everything else. The purpose of this unusual
+; design is to provide separate volume controls for voice and other audio,
+; which can be useful in gaming.
+;
+; See default.conf for an explanation on the directives used here.
+
+[General]
+auto-profiles = yes
+
+[Mapping analog-mono]
+device-strings = hw:%f,0,0
+channel-map = mono
+paths-output = steelseries-arctis-7-output-mono
+paths-input = steelseries-arctis-7-input
+
+[Mapping analog-stereo]
+device-strings = hw:%f,1,0
+channel-map = left,right
+paths-output = steelseries-arctis-7-output-stereo
+direction = output
+
+[Profile output:analog-mono+output:analog-stereo+input:analog-mono]
+output-mappings = analog-mono analog-stereo
+input-mappings = analog-mono
+priority = 5100
+skip-probe = yes
Index: pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-7-input.conf
===================================================================
--- /dev/null
+++ pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-7-input.conf
@@ -0,0 +1,25 @@
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2.1 of the
+# License, or (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
+
+; Steelseries Arctis 7 USB headset microphone path.
+
+[General]
+description-key = analog-input-microphone-headset
+
+[Element Headset]
+volume = merge
+switch = mute
+override-map.1 = all
+override-map.2 = all-left,all-right
Index: pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-7-output-mono.conf
===================================================================
--- /dev/null
+++ pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-7-output-mono.conf
@@ -0,0 +1,29 @@
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2.1 of the
+# License, or (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
+
+; Steelseries Arctis 7 USB headset mono output path. The headset has two
+; output devices. The first one is mono, meant for voice audio, and the
+; second one is stereo, meant for everything else. The purpose of this
+; unusual design is to provide separate volume controls for voice and
+; other audio, which can be useful in gaming.
+
+[General]
+description-key = analog-output-headphones-mono
+
+[Element PCM]
+volume = merge
+switch = mute
+override-map.1 = all
+override-map.2 = all-left,all-right
Index: pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-7-output-stereo.conf
===================================================================
--- /dev/null
+++ pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-7-output-stereo.conf
@@ -0,0 +1,27 @@
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2.1 of the
+# License, or (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
+
+; Steelseries Arctis 7 USB headset stereo output path. The headset has two
+; output devices. The first one is mono, meant for voice audio, and the
+; second one is stereo, meant for everything else. The purpose of this
+; unusual design is to provide separate volume controls for voice and
+; other audio, which can be useful in gaming.
+;
+; This path doesn't provide hardware volume control, because the stereo
+; output is controlled by the PCM element with index 1, and currently
+; PulseAudio only supports elements with index 0.
+
+[General]
+description-key = analog-output-headphones
Index: pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-input.conf
===================================================================
--- pulseaudio-12.2.orig/src/modules/alsa/mixer/paths/steelseries-arctis-input.conf
+++ /dev/null
@@ -1,25 +0,0 @@
-# This file is part of PulseAudio.
-#
-# PulseAudio is free software; you can redistribute it and/or modify
-# it under the terms of the GNU Lesser General Public License as
-# published by the Free Software Foundation; either version 2.1 of the
-# License, or (at your option) any later version.
-#
-# PulseAudio is distributed in the hope that it will be useful, but
-# WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
-# General Public License for more details.
-#
-# You should have received a copy of the GNU Lesser General Public License
-# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
-
-; Steelseries Arctis 7 USB headset microphone path.
-
-[General]
-description-key = analog-input-microphone-headset
-
-[Element Headset]
-volume = merge
-switch = mute
-override-map.1 = all
-override-map.2 = all-left,all-right
Index: pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-output-mono.conf
===================================================================
--- pulseaudio-12.2.orig/src/modules/alsa/mixer/paths/steelseries-arctis-output-mono.conf
+++ /dev/null
@@ -1,29 +0,0 @@
-# This file is part of PulseAudio.
-#
-# PulseAudio is free software; you can redistribute it and/or modify
-# it under the terms of the GNU Lesser General Public License as
-# published by the Free Software Foundation; either version 2.1 of the
-# License, or (at your option) any later version.
-#
-# PulseAudio is distributed in the hope that it will be useful, but
-# WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
-# General Public License for more details.
-#
-# You should have received a copy of the GNU Lesser General Public License
-# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
-
-; Steelseries Arctis 7 USB headset mono output path. The headset has two
-; output devices. The first one is mono, meant for voice audio, and the
-; second one is stereo, meant for everything else. The purpose of this
-; unusual design is to provide separate volume controls for voice and
-; other audio, which can be useful in gaming.
-
-[General]
-description-key = analog-output-headphones-mono
-
-[Element PCM]
-volume = merge
-switch = mute
-override-map.1 = all
-override-map.2 = all-left,all-right
Index: pulseaudio-12.2/src/modules/alsa/mixer/paths/steelseries-arctis-output-stereo.conf
===================================================================
--- pulseaudio-12.2.orig/src/modules/alsa/mixer/paths/steelseries-arctis-output-stereo.conf
+++ /dev/null
@@ -1,27 +0,0 @@
-# This file is part of PulseAudio.
-#
-# PulseAudio is free software; you can redistribute it and/or modify
-# it under the terms of the GNU Lesser General Public License as
-# published by the Free Software Foundation; either version 2.1 of the
-# License, or (at your option) any later version.
-#
-# PulseAudio is distributed in the hope that it will be useful, but
-# WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
-# General Public License for more details.
-#
-# You should have received a copy of the GNU Lesser General Public License
-# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
-
-; Steelseries Arctis 7 USB headset stereo output path. The headset has two
-; output devices. The first one is mono, meant for voice audio, and the
-; second one is stereo, meant for everything else. The purpose of this
-; unusual design is to provide separate volume controls for voice and
-; other audio, which can be useful in gaming.
-;
-; This path doesn't provide hardware volume control, because the stereo
-; output is controlled by the PCM element with index 1, and currently
-; PulseAudio only supports elements with index 0.
-
-[General]
-description-key = analog-output-headphones
